Userland/Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -1833,4 +1837,153 @@ IntersectionObserver::IntersectionObserverRegistration& Element::get_intersectio
     return *registration_iterator;
 }
 
+// https://html.spec.whatwg.org/multipage/dom.html#attr-dir
+Optional<Element::Dir> Element::dir_attribute() const
+{
+    auto dir = attribute(HTML::AttributeNames::dir);
+    if (dir.equals_ignoring_ascii_case("ltr"sv))
+        return Dir::Ltr;
+    if (dir.equals_ignoring_ascii_case("rtl"sv))
+        return Dir::Rtl;
+    if (dir.equals_ignoring_ascii_case("auto"sv))
+        return Dir::Auto;
+    return {};
+}
```

Comment:
Instead of scanning through the attributes whenever `dir()` is called, I suggest we keep an `Element::Dir m_dir` and update it whenever the attribute changes.

Reply:
:+1: